patroni/validator.py

```diff
@@ -56,40 +99,98 @@ def validate_host_port(host_port, listen=False, multiple_hosts=False):
+    """Validate a list of host(s) and port items.
+
+    Call :func:`validate_host_port` with each item in *value*.
+
+    :param value: list of host(s) and port items to be validated.
+    :returns: ``True`` if all items are valid.
+    """
     assert all([validate_host_port(v) for v in value]), "didn't pass the validation"
     return True
 
 
-def comma_separated_host_port(string):
+def comma_separated_host_port(string: str) -> bool:
+    """Validate a list of host and port items.
+
+    Call :func:`validate_host_port_list` with a list represented by the CSV *string*.
+
+    :param string: comma-separated list of host and port items.
+    :returns: ``True`` if all items in the CSV string are valid.
+    """
     return validate_host_port_list([s.strip() for s in string.split(",")])
 
 
-def validate_host_port_listen(host_port):
+def validate_host_port_listen(host_port: str) -> bool:
+    """Check if host and port are valid and available for binding.
+
+    Call :func:`validate_host_port` with *listen* set to ``True``.
+
+    :param host_port: the host and port to be validated. Must be in the format
+        `host:ip`.
+
+    :returns: ``True`` if the host and port are valid and available for binding.
+    """
     return validate_host_port(host_port, listen=True)
 
 
-def validate_host_port_listen_multiple_hosts(host_port):
+def validate_host_port_listen_multiple_hosts(host_port: str) -> bool:
+    """Check if host(s) and port are valid and available for binding.
+
+    Call :func:`validate_host_port` with both *listen* and *multiple_hosts* set to ``True``.
+
+    :param host_port: the host(s) and port to be validated. It can be in either of these formats
+        * `host:ip`; or
+        * `host_1,host_2,...,host_n:port`
+
+    :returns: ``True`` if the host(s) and port are valid and available for binding.
+    """
     return validate_host_port(host_port, listen=True, multiple_hosts=True)
 
 
-def is_ipv4_address(ip):
+def is_ipv4_address(ip: str) -> bool:
+    """Check if *ip* is a valid IPv4 address.
+
+    :param ip: the IP to be checked.
+    :returns: ``True`` if the IP is an IPv4 address.
+    :raises :class:`patroni.exceptions.ConfigParserError`: if *ip* is not a valid IPv4 address.
+    """
     try:
         socket.inet_aton(ip)
     except Exception:
         raise ConfigParseError("Is not a valid ipv4 address")
     return True
 
 
-def is_ipv6_address(ip):
+def is_ipv6_address(ip: str) -> bool:
+    """Check if *ip* is a valid IPv6 address.
+
+    :param ip: the IP to be checked.
+    :returns: ``True`` if the IP is an IPv6 address.
+    :raises :class:`patroni.exceptions.ConfigParserError`: if *ip* is not a valid IPv6 address.
+    """
     try:
         socket.inet_pton(socket.AF_INET6, ip)
     except Exception:
         raise ConfigParseError("Is not a valid ipv6 address")
     return True
 
 
-def get_major_version(bin_dir=None):
+def get_major_version(bin_dir: str = None) -> str:
```

Comment:
```suggestion
def get_major_version(bin_dir: OptionalType[str] = None) -> str:
```

Reply:
Besides that please also change https://github.com/zalando/patroni/blob/a04612b8fe7c40afeb55095b42ae802d39e54a54/patroni/validator.py#L199-L200 to:
```python
    match = re.match(r'^[^\s]+ [^\s]+ (\d+)(\.(\d+))?', version)
    if not match:
        return version
    return '.'.join([match.group(1), match.group(3)]) if int(match.group(1)) < 10 else match.group(1)
```

Because pyright linter complains about them.